tpu-provisioner/internal/controller/deletion_controller.go

```diff
@@ -142,17 +142,14 @@ func (r *DeletionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	// If this point is reached, the node pool has passed the deletion check twice
 	// and can be deleted.
 	lg.Info(fmt.Sprintf("Node pool %q passed deletion check twice. Ensuring Node Pool is deleted", nodePoolName))
-	r.Recorder.Event(&node, corev1.EventTypeNormal, EventDeletingNodePool, DeletingNodePoolEventMessage)
-	if err := r.Provider.DeleteNodePoolForNode(&node); err != nil {
+	if err := r.Provider.DeleteNodePoolForNode(&node, "no user Pods are running on any of the Nodes in this node pool"); err != nil {
 		if errors.Is(err, cloud.ErrDuplicateRequest) {
 			lg.Info("Ignoring duplicate request to delete node pool")
 			return ctrl.Result{}, nil
 		} else {
-			r.Recorder.Event(&node, corev1.EventTypeWarning, EventFailedDeletingNodePool, "Failed to delete Node Pool: "+err.Error())
```

Comment:
I see we are removing several multiple events we were previously emitting in different places, can you clarify why this is? Is the intent to consolidate the events to only the most useful, descriptive ones?

Also, why are we no longer ignoring the "Not found" error here?

Reply:
The not found check lives in the cloud package now. I dont think this not-found check was working b/c it checks for k8s errors not GCP errors.

Reply:
PS: On events: yes, I refactored them to provide more meaningful information. To do this they needed to be fired from the cloud pkg.